pkg/provider/gcp/secretmanager/client.go

```diff
@@ -183,13 +184,35 @@ func (c *Client) PushSecret(ctx context.Context, secret *corev1.Secret, pushSecr
 		}
 
 		if c.store.Location != "" {
+			replica := &secretmanagerpb.Replication_UserManaged_Replica{
+				Location: c.store.Location,
+			}
+
+			var meta *metadata.PushSecretMetadata[PushSecretMetadataSpec]
+			if pushSecretData.GetMetadata() != nil {
+				var err error
+				meta, err = metadata.ParseMetadataParameters[PushSecretMetadataSpec](pushSecretData.GetMetadata())
+				if err != nil {
+					return fmt.Errorf("failed to parse PushSecret metadata: %w", err)
+				}
+			}
+
+			var cmekKeyName string
+			if meta != nil {
+				cmekKeyName = meta.Spec.CMEKKeyName
+			}
+
+			if cmekKeyName != "" {
+				replica.CustomerManagedEncryption = &secretmanagerpb.CustomerManagedEncryption{
+					KmsKeyName: cmekKeyName,
+				}
+			}
```

Comment:
```suggestion
			if pushSecretData.GetMetadata() != nil {
				var err error
				meta, err := metadata.ParseMetadataParameters[PushSecretMetadataSpec](pushSecretData.GetMetadata())
				if err != nil {
					return fmt.Errorf("failed to parse PushSecret metadata: %w", err)
				}
				if meta != nil && meta.Spec.CMEKKeyName != "" {
					replica.CustomerManagedEncryption = &secretmanagerpb.CustomerManagedEncryption{
						KmsKeyName: cmekKeyName,
					}
				}
			}
```

In fact, I doubt that once metadata is defined you would end up with an empty CMEKKeyName field, right? But it's fine to check anyways.